pkg/storage/snapshot/snapshot.go

```diff
@@ -728,21 +721,32 @@ func (ctrl *VMSnapshotController) updateSnapshotStatus(vmSnapshot *snapshotv1.Vi
 		} else {
 			updateSnapshotCondition(vmSnapshotCpy, newProgressingCondition(corev1.ConditionFalse, "Source does not exist"))
 		}
-		updateSnapshotCondition(vmSnapshotCpy, newReadyCondition(corev1.ConditionFalse, "Not ready"))
+
 		if vmSnapshotDeleting(vmSnapshotCpy) {
 			vmSnapshotCpy.Status.Phase = snapshotv1.Deleting
 			updateSnapshotCondition(vmSnapshotCpy, newProgressingCondition(corev1.ConditionFalse, "VM snapshot is deleting"))
-			updateSnapshotCondition(vmSnapshotCpy, newReadyCondition(corev1.ConditionFalse, "VM snapshot is deleting"))
```

Comment:
why do we not want to update the ready condition to  false? You are changing it if phase failed and if succeeded..

Reply:
I think it's better to set ready once and simply based on whether the snapshotcontent is ready or not

Reply:
But if the vmsnapshot is deleting only on the content reconcile will the ready be changed to false and then another reconcile for the vmsnapshot to change the ready to false which is a redundant extra update if we know it will happen next reconcile, and we will show ready true although it is being deleted and shouldn't be.

Reply:
Quite an edge case I don't think we have to optimize for (if I understand correctly).  

Say someone puts their own finalizer on a succeeded vmsnapshot with "retain" policy set.  Then someone deleted the snapshot.  If someone wants to restore from that deleted snapshot why not let them?